azkaban-common/src/main/java/azkaban/executor/ExecutionControllerUtils.java

```diff
@@ -204,112 +203,71 @@ public static void restartFlow(final ExecutableFlow flow, final Status originalS
     }
   }
 
-  private static void submitRestartFlow(ExecutableFlow flow) {
+  private static void submitRestartFlow(final ExecutableFlow flow) {
     EXECUTOR_SERVICE.execute(() -> restartFlow(flow));
   }
 
   /**
    * This method invokes a callback to submit the flow with a new execution id
+   *
    * @param flow
    */
-  protected static void restartFlow(ExecutableFlow flow) {
+  protected static void restartFlow(final ExecutableFlow flow) {
     onExecutionEventListener.onExecutionEvent(flow, Constants.RESTART_FLOW);
   }
 
   /**
    * When a flow is finished, alert the user as is configured in the execution options.
    *
-   * @param flow the execution
+   * @param flow          the execution
    * @param alerterHolder the alerter holder
-   * @param extraReasons the extra reasons for alerting
+   * @param extraReasons  the extra reasons for alerting
    */
-  public static void alertUserOnFlowFinished(final ExecutableFlow flow, final AlerterHolder
-      alerterHolder, final String[] extraReasons) {
-    final ExecutionOptions options = flow.getExecutionOptions();
```

Comment:
So this diff will default to use all alerters, no longer make them optional?

Reply:
correct, control is passed to each and every configured plugin, it's for their logic to decide whether to do something or not. Knowledge of when to call a plugin isn't/shouldn't belong in Azkaban.